### ISSUE TITLE
376.0.0 ld-linux-x86-64.so.2 error

### ISSUE THREAD

**Issue by gtirloni** [GoogleCloudPlatform/cloud-sdk-docker#267]
375.0.0 works fine, 376.0.0 has this error with Alpine:

```
~ $ docker run --rm -ti alpine

/ # cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.15.0
PRETTY_NAME="Alpine Linux v3.15"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"

/ # apk --no-cache add \
>         curl \
>         python3 \
>         py3-crcmod \
>         py3-openssl \
>         bash \
>         libc6-compat \
>         openssh-client \
>         git \
>         gnupg
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
(1/59) Installing ncurses-terminfo-base (6.3_p20211120-r0)
(2/59) Installing ncurses-libs (6.3_p20211120-r0)
(3/59) Installing readline (8.1.1-r0)
(4/59) Installing bash (5.1.16-r0)
Executing bash-5.1.16-r0.post-install
(5/59) Installing ca-certificates (20211220-r0)
(6/59) Installing brotli-libs (1.0.9-r5)
(7/59) Installing nghttp2-libs (1.46.0-r0)
(8/59) Installing libcurl (7.80.0-r0)
(9/59) Installing curl (7.80.0-r0)
(10/59) Installing expat (2.4.7-r0)
(11/59) Installing pcre2 (10.39-r0)
(12/59) Installing git (2.34.1-r0)
(13/59) Installing libgpg-error (1.42-r1)
(14/59) Installing libassuan (2.5.5-r0)
(15/59) Installing libcap (2.61-r0)
(16/59) Installing pinentry (1.2.0-r0)
Executing pinentry-1.2.0-r0.post-install
(17/59) Installing libgcrypt (1.9.4-r0)
(18/59) Installing gnupg-gpgconf (2.2.31-r1)
(19/59) Installing libbz2 (1.0.8-r1)
(20/59) Installing sqlite-libs (3.36.0-r0)
(21/59) Installing gpg (2.2.31-r1)
(22/59) Installing npth (1.6-r1)
(23/59) Installing gpg-agent (2.2.31-r1)
(24/59) Installing gpg-wks-server (2.2.31-r1)
(25/59) Installing libksba (1.6.0-r0)
(26/59) Installing gpgsm (2.2.31-r1)
(27/59) Installing gpgv (2.2.31-r1)
(28/59) Installing gmp (6.2.1-r1)
(29/59) Installing nettle (3.7.3-r0)
(30/59) Installing libffi (3.4.2-r1)
(31/59) Installing p11-kit (0.24.0-r1)
(32/59) Installing libtasn1 (4.18.0-r0)
(33/59) Installing libunistring (0.9.10-r1)
(34/59) Installing gnutls (3.7.1-r0)
(35/59) Installing gdbm (1.22-r0)
(36/59) Installing libsasl (2.1.27-r14)
(37/59) Installing libldap (2.6.0-r0)
(38/59) Installing gnupg-dirmngr (2.2.31-r1)
(39/59) Installing gnupg-utils (2.2.31-r1)
(40/59) Installing gnupg-wks-client (2.2.31-r1)
(41/59) Installing gnupg (2.2.31-r1)
(42/59) Installing libc6-compat (1.2.2-r7)
(43/59) Installing openssh-keygen (8.8_p1-r1)
(44/59) Installing libedit (20210910.3.1-r0)
(45/59) Installing openssh-client-common (8.8_p1-r1)
(46/59) Installing openssh-client-default (8.8_p1-r1)
(47/59) Installing xz-libs (5.2.5-r0)
(48/59) Installing libgcc (10.3.1_git20211027-r0)
(49/59) Installing libstdc++ (10.3.1_git20211027-r0)
(50/59) Installing mpdecimal (2.5.1-r1)
(51/59) Installing python3 (3.9.7-r4)
(52/59) Installing py3-crcmod (1.7-r5)
(53/59) Installing py3-cparser (2.20-r1)
(54/59) Installing py3-cffi (1.14.5-r2)
(55/59) Installing py3-idna (3.3-r0)
(56/59) Installing py3-asn1crypto (1.4.0-r1)
(57/59) Installing py3-six (1.16.0-r0)
(58/59) Installing py3-cryptography (3.3.2-r3)
(59/59) Installing py3-openssl (21.0.0-r0)
Executing busybox-1.34.1-r3.trigger
Executing ca-certificates-20211220-r0.trigger
OK: 93 MiB in 73 packages

/ # curl -O https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-376.0.0-linux-x86_64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 89.7M  100 89.7M    0     0  32.5M      0  0:00:02  0:00:02 --:--:-- 32.5M

/ # tar xfz google-cloud-sdk-376.0.0-linux-x86_64.tar.gz

/ # /google-cloud-sdk/install.sh
Welcome to the Google Cloud CLI!
WARNING: You appear to be running this script as root. This may cause 
the installation to be inaccessible to users other than the root user.
Traceback (most recent call last):
  File "/google-cloud-sdk/bin/bootstrapping/install.py", line 12, in <module>
    import bootstrapping
  File "/google-cloud-sdk/bin/bootstrapping/bootstrapping.py", line 46, in <module>
    from googlecloudsdk.core.updater import update_manager
  File "/google-cloud-sdk/lib/googlecloudsdk/core/updater/update_manager.py", line 40, in <module>
    from googlecloudsdk.core.resource import resource_printer
  File "/google-cloud-sdk/lib/googlecloudsdk/core/resource/resource_printer.py", line 42, in <module>
    from googlecloudsdk.core.resource import config_printer
  File "/google-cloud-sdk/lib/googlecloudsdk/core/resource/config_printer.py", line 25, in <module>
    from googlecloudsdk.core.resource import resource_printer_base
  File "/google-cloud-sdk/lib/googlecloudsdk/core/resource/resource_printer_base.py", line 47, in <module>
    from googlecloudsdk.core.resource import resource_projector
  File "/google-cloud-sdk/lib/googlecloudsdk/core/resource/resource_projector.py", line 71, in <module>
    import proto  # pylint: disable=g-import-not-at-top
  File "/google-cloud-sdk/lib/third_party/proto/__init__.py", line 15, in <module>
    from .enums import Enum
  File "/google-cloud-sdk/lib/third_party/proto/enums.py", line 17, in <module>
    from google.protobuf import descriptor_pb2
  File "/google-cloud-sdk/lib/third_party/google/protobuf/descriptor_pb2.py", line 5, in <module>
    from google.protobuf import descriptor as _descriptor
  File "/google-cloud-sdk/lib/third_party/google/protobuf/descriptor.py", line 48, in <module>
    from google.protobuf.pyext import _message
ImportError: Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by /google-cloud-sdk/lib/third_party/google/protobuf/pyext/_message.cpython-39-x86_64-linux-gnu.so)


/ # apk add --no-cache gcompat
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
(1/3) Installing musl-obstack (1.2.2-r0)
(2/3) Installing libucontext (1.1-r0)
(3/3) Installing gcompat (1.0.0-r4)
OK: 93 MiB in 76 packages

/ # /google-cloud-sdk/install.sh
Welcome to the Google Cloud CLI!
WARNING: You appear to be running this script as root. This may cause 
the installation to be inaccessible to users other than the root user.
terminate called after throwing an instance of 'std::system_error'
  what():  No error information
Aborted (core dumped)


```

**Comment by dinapappor** [comment]
When I try I only see this working with alpine `3.13` or older. 3.14 or 3.15 fails with `ImportError: Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by /google-cloud-sdk/lib/third_party/google/protobuf/pyext/_message.cpython-39-x86_64-linux-gnu.so)`

**Comment by dinapappor** [comment]
```
FROM docker:20.10.12 as static-docker-source

FROM alpine:3.15
ARG CLOUD_SDK_VERSION=375.0.0
ENV CLOUD_SDK_VERSION=$CLOUD_SDK_VERSION
ENV PATH /google-cloud-sdk/bin:$PATH
COPY --from=static-docker-source /usr/local/bin/docker /usr/local/bin/docker
RUN addgroup -g 1000 -S cloudsdk && \
    adduser -u 1000 -S cloudsdk -G cloudsdk
RUN apk --no-cache add \
        curl \
        python3 \
        py3-crcmod \
        py3-openssl \
        bash \
        libc6-compat \
        openssh-client \
        git \
        gnupg \
    && curl -O https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-${CLOUD_SDK_VERSION}-linux-x86_64.tar.gz && \
    tar xzf google-cloud-sdk-${CLOUD_SDK_VERSION}-linux-x86_64.tar.gz && \
    rm google-cloud-sdk-${CLOUD_SDK_VERSION}-linux-x86_64.tar.gz && \
    gcloud config set core/disable_usage_reporting true && \
    gcloud config set component_manager/disable_update_check true && \
    gcloud config set metrics/environment github_docker_image && \
    gcloud --version \
    && /google-cloud-sdk/install.sh --override-components alpha beta bq core gsutil kubectl
RUN git config --system credential.'https://source.developers.google.com'.helper gcloud.sh
VOLUME ["/root/.config"]
```

This works for me with 3.15. But not with 376

**Comment by gtirloni** [comment]
I think it's the usual [native Python extension](https://pypi.org/project/protobuf) is not compiled for whatever version of musl that Alpine 3.14/3.15 is using now and/or the libc6-compat/gcompat is not enough anymore.

**Comment by s4shyam95** [comment]
This will be fixed in the 378.0.0 release.

**Comment by willfaris** [comment]
378.0.0 has been released.
